lib/deployer.rb

```diff
@@ -0,0 +1,42 @@
+require "rake/file_utils"
+
+class Deployer
+  include FileUtils
+
+  attr_reader :build_dir, :deploy_branch, :deploy_sha
+
+  def self.prepare_deploy(options={})
+    self.new(options).prepare_deploy
+  end
+
+  def initialize(options={})
+    @build_dir = options.fetch(:build_dir, "_site")
+    @deploy_branch = options.fetch(:deploy_branch, "gh-pages")
+    @deploy_sha = options.fetch(:deploy_sha)
+  end
+
+  def prepare_deploy
+    build
+    prepare_deploy_branch
+    promote_build
+  end
```

Comment:
So - I'm sure this happens super fast but it could be nice for the person running the rake task to get some feedback about what's happening. Then also you could have the rake task prompt the user's next steps so that they don't have to flip back to the README.